theme/js/index.js

```diff
@@ -0,0 +1,126 @@
+// Mobile nav controls
+const navButton = document.querySelector('.nav__menu--button');
+const closeButton = document.querySelector('.nav__menu--close');
+const mobileNavMenu = document.querySelector('.k-nav');
+const pageContainer = document.querySelector('.page__container');
+
+navButton.addEventListener('click', () => {
+  mobileNavMenu.style.display = 'block';
+  closeButton.style.display = 'block';
+  navButton.style.display = 'none';
+  pageContainer.style.position = 'fixed';
+});
+
+closeButton.addEventListener('click', () => {
+  mobileNavMenu.style.display = 'none';
+  closeButton.style.display = 'none';
+  navButton.style.display = 'block';
+  pageContainer.style.position = 'static';
+});
+
+// Copy code
+const copyButtons = document.querySelectorAll('.code__copy--button');
+copyButtons.forEach((button) => {
+  button.addEventListener('click', () => {
+    const parent = button.parentNode;
+    const text = parent.querySelector('.language-python').innerText;
+    const inputElement = document.createElement('textarea');
+    console.log('text', text);
+    inputElement.value = text;
+    inputElement.setAttribute('class', 'visually-hidden');
+    const body = document.body;
+    body.appendChild(inputElement);
+    inputElement.select();
+    document.execCommand('copy');
+    inputElement.remove();
+  });
+});
+
+// Search controls
+const searchForms = document.querySelectorAll('.nav__search');
+const mobileNavSearchIcon = document.querySelector('.nav__search--mobile');
+const mobileNavSearchForm = document.querySelector('.nav__search-form--mobile');
+const mobileNavControls = document.querySelector('.nav__controls--mobile');
+
+mobileNavSearchIcon.addEventListener('click', () => {
+  mobileNavControls.style.display = 'none';
+  mobileNavSearchForm.style.display = 'block';
+});
+
+searchForms.forEach((search) => {
+  search.addEventListener('submit', (event) => {
+    event.preventDefault();
+    const text = search.querySelector('.nav__search--input').value;
+    window.location = `/search.html?query=${text}`;
+  });
+});
+
+pageContainer.addEventListener('click', () => {
+  mobileNavControls.style.display = 'flex';
+  mobileNavSearchForm.style.display = 'none';
+});
+
+mobileNavMenu.addEventListener('click', () => {
+  mobileNavControls.style.display = 'flex';
+  mobileNavSearchForm.style.display = 'none';
+});
+
+// Parallax functionality
```

Comment:
I wonder if this whole code block would be less jerky if you dynamically updated the element css to switch between position fixed and relative--so the render engine will keep fixed rather than trying to dynamically set a height in js. Not a web dev expert though!